streams/src/main/java/org/apache/kafka/streams/state/internals/ThreadCache.java

```diff
@@ -101,7 +101,7 @@ public static String taskIDfromCacheName(final String cacheName) {
      * @return
      */
     public static String underlyingStoreNamefromCacheName(final String cacheName) {
-        String[] tokens = cacheName.split("-");
+        String[] tokens = cacheName.split("-", 2);
```

Comment:
`cacheName` is created in `nameSpaceFromTaskIdAndStore` in L84 by concatenating the `taskID` and the `underlyingStoreName` with a `"-"`. So when we take them back apart, we should just split on hyphen *once*. Previously, this would split on *every* hyphen, returning only the part of the store name up to the first hyphen. This is a problem since store names may contain hyphens.

The result is that all named caches for stores with the same prefix (like "agg-store" and "agg-STATE_STORE_000000") will both try to create metrics with a storeName of just "agg".

We never noticed before because all NamedCaches were actually just creating one sensor, and it's ok to register duplicate metrics on the same sensor. Once I split the sensors up, it threw an exception on the duplicate metrics.

Reply:
seems reasonable to me